starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,51 @@
+const express = require("express");
+const authRoutes = express.Router();
+
+authRoutes.get("/signup", (req, res, next) => {
+    res.render("auth/signup");
+});
+const User = require("../models/user");
+
+const bcrypt = require("bcryptjs");
+const bcryptSalt = 14;
+
+authRoutes.post("/signup", (req, res, next) => {
+    var username = req.body.username;
+    var password = req.body.password;
+    console.log(username,password);
+    bcrypt.genSalt(bcryptSalt, (err, salt) => {
+        bcrypt.hash(password, salt, (err, hash) => {
+
+            var newUser = User({
+                username,
+                password: hash
+            });
+        
+            newUser.save((err) => {
+                if (username === "" || password === "") {
+                    res.render("auth/signup", {
+                        errorMessage: "Indicate a username and a password to sign up"
+                    });
+                    res.redirect("/");
```

Comment:
the `res.redirect("/")` is inside the `if` body, so actually the user is never redirected. That's why your signup route is working properly with saving a new user to the db, but the browser hangs and doesn't do anything after that. If you move it outside the `if`, it will work.